thrust/testing/type_traits.cu

```diff
@@ -146,3 +152,32 @@ void TestIsCommutative()
   }
 }
 DECLARE_UNITTEST(TestIsCommutative);
+
+void TestTriviallyRelocatable()
+{
+  static_assert(thrust::is_trivially_relocatable<int>::value, "");
+#if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
+  static_assert(thrust::is_trivially_relocatable<__half>::value, "");
+  static_assert(thrust::is_trivially_relocatable<int1>::value, "");
+  static_assert(thrust::is_trivially_relocatable<int2>::value, "");
+  static_assert(thrust::is_trivially_relocatable<int3>::value, "");
+  static_assert(thrust::is_trivially_relocatable<int4>::value, "");
+  static_assert(thrust::is_trivially_relocatable<__int128>::value, "");
```

Comment:
This needs to be guarded by availability of `__int128_t`

We have slightly different definitions for those in cub and libcu++. Choose wisely ;)

Reply:
I take libcu++.

Reply:
Why would I want to have a `_CCCL_HAS_INT128`, if everything depends on libcu++ anyway? I would just accumulate all configuration there.

Reply:
The reasone is 
a) libcu++ has the double negation from libc++ which is bad
b) we want to move global macro definitions into cccl because they have nothing to do with libcu++. compilers and standard modes are example of such macros. Availability of int128 definitely is one too

Reply:
> we want to move global macro definitions into cccl because they have nothing to do with libcu++.

This is news to me, I thought libcu++ was at the root of food chain.

Anyhow, I don't intend to fix this within this PR.